controllers/tasks.js

```diff
@@ -16,17 +17,23 @@ const { INTERNAL_SERVER_ERROR, SOMETHING_WENT_WRONG } = require("../constants/er
 const addNewTask = async (req, res) => {
   try {
     const { id: createdBy } = req.userData;
+    const dependsOn = req.body.dependsOn;
     const body = {
       ...req.body,
       createdBy,
     };
-
-    const task = await tasks.updateTask(body);
-
+    delete body.dependsOn;
+    const { taskId, taskDetails } = await tasks.updateTask(body);
+    const data = {
+      taskId,
+      dependsOn,
+    };
+    const taskDependency = await dependencyModel.addDependency(data);
     return res.json({
       message: "Task created successfully!",
-      task: task.taskDetails,
-      id: task.taskId,
+      task: taskDetails,
+      id: taskId,
+      dependsOn: taskDependency,
```

Comment:
Nit : `dependsOn` should be part of the taskDetails rather than directly being at the parent level.

Reply:
resolved